src/components/data_display/helpers.ts

```diff
@@ -92,7 +97,13 @@ export const buildAllColumnDefinitions = async ({
   const columnsToReturn: string[] = buildColumnList(dataView, fieldNames, type)
 
   const filterColumns: string[] =
-    type === 'TABLE' ? buildColumnList(dataView, fieldNames, 'FILTER') : []
+    type === 'TABLE'
+      ? buildColumnList(dataView, fieldNames, 'FILTER').map((column: string) =>
+          fieldNames.includes(`${column}${FILTER_COLUMN_SUFFIX}`)
+            ? `${column}${FILTER_COLUMN_SUFFIX}`
+            : column
+        )
+      : []
```

Comment:
Replaces normal columns with their "filterData" replacement for filter column list.